tripy/nvtripy/frontend/tensor.py

```diff
@@ -177,12 +176,24 @@ def ndim(self):
     def device(self):
         return self.trace_tensor.device
 
-    # TODO (pranavm): Document this and `tolist()`? Figure out more ergonomic way to get scalar value.
-    def eval(self) -> runtime.MemRefValue:
+    def eval(self) -> "nvtripy.Tensor":
+        """
+        Evaluates the tensor, making the tensor to be a constant.
```

Comment:
Maybe something like:
```
Immediately evaluates this tensor. By default, tensors are evaluated lazily. 
```
There's not really a notion of a "constant" in the user API, e.g. `tp.ones(...)` looks like a constant but isn't actually one.